src/SourceLocator/Type/SingleFileSourceLocator.php

```diff
@@ -26,9 +28,9 @@ class SingleFileSourceLocator extends AbstractSourceLocator
      * @param string $filename
      * @throws \Roave\BetterReflection\SourceLocator\Exception\InvalidFileLocation
      */
-    public function __construct(string $filename)
+    public function __construct(string $filename, ?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.